sdks/java/io/file-schema-transform/src/test/java/org/apache/beam/sdk/io/fileschematransform/AvroReadSchemaTransformFormatProviderTest.java

```diff
@@ -0,0 +1,265 @@
+
+import java.util.Arrays;
+import java.util.List;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.beam.sdk.extensions.avro.coders.AvroGenericCoder;
+import org.apache.beam.sdk.extensions.avro.io.AvroIO;
+import org.apache.beam.sdk.extensions.avro.io.DynamicAvroDestinations;
+import org.apache.beam.sdk.extensions.avro.schemas.utils.AvroUtils;
+import org.apache.beam.sdk.io.DefaultFilenamePolicy;
+import org.apache.beam.sdk.io.FileBasedSink;
+import org.apache.beam.sdk.io.FileSystems;
+import org.apache.beam.sdk.io.GenerateSequence;
+import org.apache.beam.sdk.io.fs.ResourceId;
+import org.apache.beam.sdk.options.ValueProvider.StaticValueProvider;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.Schema.Field;
+import org.apache.beam.sdk.schemas.Schema.FieldType;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransform;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.transforms.windowing.AfterPane;
+import org.apache.beam.sdk.transforms.windowing.FixedWindows;
+import org.apache.beam.sdk.transforms.windowing.Repeatedly;
+import org.apache.beam.sdk.transforms.windowing.Window;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+import org.apache.beam.sdk.values.Row;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.beam.sdk.values.TypeDescriptors;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Strings;
+import org.joda.time.Duration;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class AvroReadSchemaTransformFormatProviderTest
+    extends FileReadSchemaTransformFormatProviderTest {
+
+  @Rule public TestPipeline writePipeline = TestPipeline.create();
+  @Rule public TestPipeline readPipeline = TestPipeline.create();
+
+  @Override
+  protected String getFormat() {
+    return new AvroReadSchemaTransformFormatProvider().identifier();
+  }
+
+  @Override
+  public String getStringSchemaFromBeamSchema(Schema beamSchema) {
+    return AvroUtils.toAvroSchema(beamSchema).toString();
+  }
+
+  @Override
+  public void runWriteAndReadTest(
+      Schema schema, List<Row> rows, String filePath, String schemaFilePath) {
+    org.apache.avro.Schema avroSchema = AvroUtils.toAvroSchema(schema);
+    String stringSchema =
+        Strings.isNullOrEmpty(schemaFilePath) ? avroSchema.toString() : schemaFilePath;
+
+    writePipeline
+        .apply(Create.of(rows).withRowSchema(schema))
+        .apply(
+            MapElements.into(TypeDescriptor.of(GenericRecord.class))
+                .via(AvroUtils.getRowToGenericRecordFunction(avroSchema)))
+        .setCoder(AvroGenericCoder.of(avroSchema))
+        .apply(AvroIO.writeGenericRecords(avroSchema).to(filePath));
```

Comment:
Should we test whether files written using FileWriteSchemaTransformProvider read from FileReadSchemaTransformProvider?

Reply:
Yeah probably worth doing

Reply:
Added the tests